### ISSUE TITLE
fs::remove_all cannot delete non-directory files

### ISSUE THREAD

**Issue by actboy168** [gulrak/filesystem#6]
**Describe the bug**
fs::remove_all should allow deletion of a file.

**To Reproduce**
```
assert(fs::remove_all("test.txt") == 1)
assert(fs::remove_all("test.txt") == 0)
```

**Expected behavior**
But now it will throw an exception.

**Additional context**

**Comment by gulrak** [comment]
Indeed, thanks for reporting! I will push a fix on master when I'm back home and I scheduled a release for next sunday.

**Comment by gulrak** [comment]
True, added a test and a fix for that.

**Comment by gulrak** [comment]
Closed with release of v1.0.8
